### PR TITLE
Add checker for data values that don't have a group name

### PR DESCRIPTION
**Pull request by sdbernard**
This is used to set the colour palette

### CODE REVIEW

File: slope/index.js [hunk 1]
```diff
@@ -130,6 +132,7 @@ parseData.fromCSV('./data.csv', dateStructure).then(({ seriesNames, groupNames,
         myChart.dotRadius(currentFrame.rem() * 0.3);
 
         // specifics based on frame type
+        myChart.setPalette(setColourPalette);
         myChart.colourPalette(frameName, groupNames); // set colour palette
```
Comment (aendrew) [inline slope/index.js]: This is a bit confusing — you set palette, then you set colour palette? It's not a very intuitive API if both these steps are needed separately.

File: slope/parseData.js [hunk 2]
```diff
@@ -34,8 +34,16 @@ export function fromCSV(url, dateStructure) {
                     return -1;
                 };
 
+                function hasGroupName(element, index, array) {
+                  return element != '';
```
Comment (aendrew) [inline slope/parseData.js]: Should be double equals.

File: slope/index.js [hunk 3]
```diff
@@ -76,13 +76,15 @@ d3.selectAll('.framed')
...
 
-parseData.fromCSV('./data.csv', dateStructure).then(({ seriesNames, groupNames, dataSorter, data }) => {
+parseData.fromCSV('./data.csv', dateStructure).then(({ seriesNames, setColourPalette, groupNames, dataSorter, data }) => {
     // Use the seriesNames array to calculate the minimum and max values in the dataset
     const valueExtent = parseData.extentMulti(data, seriesNames);
     data.sort(dataSorter);
 
     const valueFormat = d => d3.format(',')(d);
 
+    console.log(setColourPalette);
```
Comment (aendrew) [inline slope/index.js]: Extraneous console.log.

File: slope/index.js (hunk 1 above)
Comment (aendrew) [inline slope/index.js]: Ah, I see now what you're doing — setPalette takes a boolean, and short-circuits colourPalette. Maybe it should just be passed to `.colourPalette()` as an argument?

Also, is it too late to rename "colourPalette" to "colorPalette"?

Comment (sdbernard) [inline slope/index.js]: yup that's totally doable re passing in the argument. Changing colour to color should be a problem I'd just need to find and replace in all folders

Comment (sdbernard) [inline slope/index.js]: sorry shouldn't be a problem!

Comment (sdbernard) [inline slope/index.js]: there's colourInverse as well to change

**Pull request merged**